Gradle plugin doesn't support multi module project well.

**Describe the bug**
I want to use quarkus project as sub module project. But the plugin is not working well.

**Expected behavior**
It should work with multi module project.

**Actual behavior**
https://github.com/SetoKaiba/blueeyes
Here's the test multi module project.
1. It does not recognize the gradle.properties file in the root project. It will work with placing it in the sub module. I use the subprojects DSL to simplify the configuration. So I have to duplicate the properties in parent and sub moudules.
```
#Gradle properties
#Sat Nov 23 16:34:25 UTC 2019
quarkusPluginVersion=1.0.0.CR2
quarkusPlatformArtifactId=quarkus-universe-bom
quarkusPlatformVersion=1.0.0.CR2
quarkusPlatformGroupId=io.quarkus
```

```
subprojects{
    // this block is necessary to make enforcedPlatform work for Quarkus plugin available
    // only locally (snapshot) that is also importing the Quarkus BOM
    buildscript {
        repositories {
            mavenLocal()
            maven {
                url "https://repository.jboss.org/nexus/content/groups/public-jboss"
            }
            jcenter()
        }
        dependencies {
            classpath "io.quarkus:quarkus-gradle-plugin:${quarkusPluginVersion}"
        }
    }
}
```

2. gradlew :blueeyes-web:addExtension --extensions="io.quarkus:quarkus-hibernate-orm"
It will generate settings.gradle in sub module. It will cause problems for sub module.
The problem is below after settings.gradle is generated. I have to delete it manually to avoid the problem.

```
14:37:22: Executing task 'listExtensions'...


FAILURE: Build failed with an exception.

* Where:
Build file 'D:\Quarkus\blueeyes\blueeyes-web\build.gradle' line: 1

* What went wrong:
A problem occurred evaluating root project 'blueeyes-web'.
> Plugin with id 'io.quarkus' not found.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
14:37:23: Task execution finished 'listExtensions'.
```

3. Argument in IDEA is not working. It seems a bug of IntelliJ IDEA. 
https://youtrack.jetbrains.com/issue/IDEA-227496
https://youtrack.jetbrains.com/issue/IDEA-196952

**To Reproduce**
Steps to reproduce the behavior:
1. 
2. 
3. 

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 

**Additional context**
(Add any other context about the problem here.)

--- a/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/AddExtensionToModuleInMultiModuleProjectTest.java
+++ b/devtools/gradle/src/functionalTest/java/io/quarkus/gradle/AddExtensionToModuleInMultiModuleProjectTest.java
@@ -1,0 +1,37 @@
+package io.quarkus.gradle;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.bootstrap.util.IoUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class AddExtensionToModuleInMultiModuleProjectTest extends QuarkusGradleTestBase {
+
+    @Test
+    public void testBasicMultiModuleBuild() throws Exception {
+
+        final File projectDir = getProjectDir("add-extension-multi-module");
+        
+        BuildResult build = GradleRunner.create()
+                .forwardOutput()
+                .withPluginClasspath()
+                .withArguments(arguments(":application:addExtension", "--extensions=hibernate-orm"))
+                .withProjectDir(projectDir)
+                .build();
+        
+        final Path applicationLib = projectDir.toPath().resolve("application").resolve("settings.gradle");
+        assertThat(applicationLib).doesNotExist();
+        
+        final Path appBuild = projectDir.toPath().resolve("application").resolve("build.gradle");
+        assertThat(appBuild).exists();
+        assertThat(IoUtils.readFile(appBuild)).contains("implementation 'io.quarkus:quarkus-hibernate-orm'");
+    }
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/build.gradle
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/build.gradle
@@ -1,0 +1,13 @@
+plugins {
+    id 'io.quarkus'
+}
+
+dependencies {
+    implementation project(":common")
+    implementation 'io.quarkus:quarkus-resteasy'
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured'
+
+    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/java/org/acme/quarkus/sample/HelloResource.java
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/java/org/acme/quarkus/sample/HelloResource.java
@@ -1,0 +1,22 @@
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.acme.common.CommonBean;
+
+@Path("/hello")
+public class HelloResource {
+
+    @Inject
+    CommonBean common;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello " + common.getName();
+    }
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/resources/META-INF/resources/index.html
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/resources/META-INF/resources/index.html
@@ -1,0 +1,152 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>my-quarkus-project - 1.0-SNAPSHOT</title>
+    <style>
+        h1, h2, h3, h4, h5, h6 {
+            margin-bottom: 0.5rem;
+            font-weight: 400;
+            line-height: 1.5;
+        }
+
+        h1 {
+            font-size: 2.5rem;
+        }
+
+        h2 {
+            font-size: 2rem
+        }
+
+        h3 {
+            font-size: 1.75rem
+        }
+
+        h4 {
+            font-size: 1.5rem
+        }
+
+        h5 {
+            font-size: 1.25rem
+        }
+
+        h6 {
+            font-size: 1rem
+        }
+
+        .lead {
+            font-weight: 300;
+            font-size: 2rem;
+        }
+
+        .banner {
+            font-size: 2.7rem;
+            margin: 0;
+            padding: 2rem 1rem;
+            background-color: #00A1E2;
+            color: white;
+        }
+
+        body {
+            margin: 0;
+            font-family: -apple-system, system-ui, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+        }
+
+        code {
+            font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
+            font-size: 87.5%;
+            color: #e83e8c;
+            word-break: break-word;
+        }
+
+        .left-column {
+            padding: .75rem;
+            max-width: 75%;
+            min-width: 55%;
+        }
+
+        .right-column {
+            padding: .75rem;
+            max-width: 25%;
+        }
+
+        .container {
+            display: flex;
+            width: 100%;
+        }
+
+        li {
+            margin: 0.75rem;
+        }
+
+        .right-section {
+            margin-left: 1rem;
+            padding-left: 0.5rem;
+        }
+
+        .right-section h3 {
+            padding-top: 0;
+            font-weight: 200;
+        }
+
+        .right-section ul {
+            border-left: 0.3rem solid #00A1E2;
+            list-style-type: none;
+            padding-left: 0;
+        }
+
+    </style>
+</head>
+<body>
+
+<div class="banner lead">
+    Your new Cloud-Native application is ready!
+</div>
+
+<div class="container">
+    <div class="left-column">
+        <p class="lead"> Congratulations, you have created a new Quarkus application.</p>
+
+        <h2>Why do you see this?</h2>
+
+        <p>This page is served by Quarkus. The source is in
+            <code>src/main/resources/META-INF/resources/index.html</code>.</p>
+
+        <h2>What can I do from here?</h2>
+
+        <p>If not already done, run the application in <em>dev mode</em> using: <code>mvn compile quarkus:dev</code>.
+        </p>
+        <ul>
+            <li>Add REST resources, Servlets, functions and other services in <code>src/main/java</code>.</li>
+            <li>Your static assets are located in <code>src/main/resources/META-INF/resources</code>.</li>
+            <li>Configure your application in <code>src/main/resources/application.properties</code>.
+            </li>
+        </ul>
+
+        <h2>How do I get rid of this page?</h2>
+        <p>Just delete the <code>src/main/resources/META-INF/resources/index.html</code> file.</p>
+    </div>
+    <div class="right-column">
+        <div class="right-section">
+            <h3>Application</h3>
+            <ul>
+                <li>GroupId: org.acme.quarkus.sample</li>
+                <li>ArtifactId: my-quarkus-project</li>
+                <li>Version: 1.0-SNAPSHOT</li>
+                <li>Quarkus Version: 999-SNAPSHOT</li>
+            </ul>
+        </div>
+        <div class="right-section">
+            <h3>Next steps</h3>
+            <ul>
+                <li><a href="https://quarkus.io/guides/maven-tooling.html" target="_blank">Setup your IDE</a></li>
+                <li><a href="https://quarkus.io/guides/getting-started.html" target="_blank">Getting started</a></li>
+                <li><a href="https://quarkus.io" target="_blank">Quarkus Web Site</a></li>
+            </ul>
+        </div>
+    </div>
+</div>
+
+
+</body>
+</html>

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/resources/application.properties
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/application/src/main/resources/application.properties
@@ -1,0 +1,2 @@
+# Configuration file
+# key = value

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/build.gradle
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/build.gradle
@@ -1,0 +1,39 @@
+buildscript {
+
+    repositories {
+        jcenter()
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+
+}
+
+apply plugin: 'java'
+
+group = 'com.quarkus.demo'
+version = '1.0'
+
+
+subprojects {
+
+    apply plugin: 'java'
+
+    java {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    test {
+        dependsOn 'cleanTest'
+        dependsOn 'quarkusTestConfig'
+        useJUnitPlatform()
+        forkEvery 1
+    }
+
+    repositories {
+        jcenter()
+        mavenLocal()
+        mavenCentral()
+    }
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/build.gradle
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/build.gradle
@@ -1,0 +1,7 @@
+plugins {
+    id 'java-library'
+}
+
+dependencies {
+
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/src/main/java/org/acme/common/CommonBean.java
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/src/main/java/org/acme/common/CommonBean.java
@@ -1,0 +1,11 @@
+package org.acme.common;
+
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class CommonBean {
+
+   public String getName() {
+       return "common";
+   }
+}

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/src/main/resources/META-INF/beans.xml
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/common/src/main/resources/META-INF/beans.xml
@@ -1,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans
+        xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee 
+                      http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd"
+        bean-discovery-mode="all">
+
+</beans>

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/gradle.properties
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/gradle.properties
@@ -1,0 +1,2 @@
+quarkusPlatformArtifactId=quarkus-bom
+quarkusPlatformGroupId=io.quarkus

--- a/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/settings.gradle
+++ b/devtools/gradle/src/functionalTest/resources/add-extension-multi-module/settings.gradle
@@ -1,0 +1,18 @@
+pluginManagement {
+    repositories {
+        jcenter()
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    //noinspection GroovyAssignabilityCheck
+    plugins {
+        id 'io.quarkus' version "${quarkusPluginVersion}"
+    }
+}
+
+rootProject.name = 'quarkus-basic-multi-module-build'
+
+include ':common'
+include ':application'
+

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusAddExtension.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusAddExtension.java
@@ -42,7 +42,11 @@ public class QuarkusAddExtension extends QuarkusPlatformTask {
                 .collect(toSet());
 
         try {
-            new AddExtensions(new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir())), platformDescriptor())
+            GradleBuildFile gradleBuildFile = getProject().getParent() == null
+                    ? new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()))
+                    : new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()),
+                            new FileProjectWriter(getProject().getRootProject().getProjectDir()));
+            new AddExtensions(gradleBuildFile, platformDescriptor())
                     .extensions(extensionsSet)
                     .execute();
         } catch (Exception e) {

--- a/independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java
+++ b/independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java
@@ -22,19 +22,43 @@ public class GradleBuildFile extends BuildFile {
     private static final String SETTINGS_GRADLE_PATH = "settings.gradle";
     private static final String GRADLE_PROPERTIES_PATH = "gradle.properties";
 
+    private final ProjectWriter rootWriter;
+
     private Model model;
 
     public GradleBuildFile(ProjectWriter writer) {
         super(writer, BuildTool.GRADLE);
+        rootWriter = writer;
+    }
+
+    public GradleBuildFile(ProjectWriter writer, ProjectWriter rootWriter) {
+        super(writer, BuildTool.GRADLE);
+        this.rootWriter = rootWriter;
+    }
+
+    protected void rootWrite(String fileName, String content) throws IOException {
+        rootWriter.write(fileName, content);
+    }
+
+    protected ProjectWriter getRootWriter() {
+        return rootWriter;
     }
 
     @Override
     public void close() throws IOException {
-        write(SETTINGS_GRADLE_PATH, getModel().getSettingsContent());
+        if (getWriter() == getRootWriter()) {
+            write(SETTINGS_GRADLE_PATH, getModel().getSettingsContent());
+        } else {
+            rootWrite(SETTINGS_GRADLE_PATH, getModel().getSettingsContent());
+        }
         write(BUILD_GRADLE_PATH, getModel().getBuildContent());
         try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
             getModel().getPropertiesContent().store(out, "Gradle properties");
-            write(GRADLE_PROPERTIES_PATH, out.toString(StandardCharsets.UTF_8.toString()));
+            if (getModel().isRootProperties()) {
+                rootWrite(GRADLE_PROPERTIES_PATH, out.toString(StandardCharsets.UTF_8.toString()));
+            } else {
+                write(GRADLE_PROPERTIES_PATH, out.toString(StandardCharsets.UTF_8.toString()));
+            }
         }
     }
 
@@ -208,9 +232,17 @@ public class GradleBuildFile extends BuildFile {
         String settingsContent = "";
         String buildContent = "";
         Properties propertiesContent = new Properties();
-        if (getWriter().exists(SETTINGS_GRADLE_PATH)) {
-            final byte[] settings = getWriter().getContent(SETTINGS_GRADLE_PATH);
-            settingsContent = new String(settings, StandardCharsets.UTF_8);
+        boolean isRootProperties = false;
+        if (getWriter() == getRootWriter()) {
+            if (getWriter().exists(SETTINGS_GRADLE_PATH)) {
+                final byte[] settings = getWriter().getContent(SETTINGS_GRADLE_PATH);
+                settingsContent = new String(settings, StandardCharsets.UTF_8);
+            }
+        } else {
+            if (getRootWriter().exists(SETTINGS_GRADLE_PATH)) {
+                final byte[] settings = getRootWriter().getContent(SETTINGS_GRADLE_PATH);
+                settingsContent = new String(settings, StandardCharsets.UTF_8);
+            }
         }
         if (getWriter().exists(BUILD_GRADLE_PATH)) {
             final byte[] build = getWriter().getContent(BUILD_GRADLE_PATH);
@@ -220,7 +252,17 @@ public class GradleBuildFile extends BuildFile {
             final byte[] properties = getWriter().getContent(GRADLE_PROPERTIES_PATH);
             propertiesContent.load(new ByteArrayInputStream(properties));
         }
-        this.model = new Model(settingsContent, buildContent, propertiesContent);
+        if (!propertiesContent.containsKey("quarkusPluginVersion") &&
+                !propertiesContent.containsKey("quarkusPlatformGroupId") &&
+                !propertiesContent.containsKey("quarkusPlatformArtifactId") &&
+                !propertiesContent.containsKey("quarkusPlatformVersion")) {
+            if (getRootWriter().exists(GRADLE_PROPERTIES_PATH)) {
+                final byte[] properties = getRootWriter().getContent(GRADLE_PROPERTIES_PATH);
+                propertiesContent.load(new ByteArrayInputStream(properties));
+            }
+            isRootProperties = true;
+        }
+        this.model = new Model(settingsContent, buildContent, propertiesContent, isRootProperties);
     }
 
     protected String getBuildContent() throws IOException {
@@ -231,11 +273,13 @@ public class GradleBuildFile extends BuildFile {
         private String settingsContent;
         private String buildContent;
         private Properties propertiesContent;
+        private boolean rootProperties;
 
-        public Model(String settingsContent, String buildContent, Properties propertiesContent) {
+        public Model(String settingsContent, String buildContent, Properties propertiesContent, boolean rootProperties) {
             this.settingsContent = settingsContent;
             this.buildContent = buildContent;
             this.propertiesContent = propertiesContent;
+            this.rootProperties = rootProperties;
         }
 
         public String getSettingsContent() {
@@ -256,6 +300,10 @@ public class GradleBuildFile extends BuildFile {
 
         public void setBuildContent(String buildContent) {
             this.buildContent = buildContent;
+        }
+
+        public boolean isRootProperties() {
+            return rootProperties;
         }
     }
 }

